From ea7a6ef806ac1b148c17d016e4effa6a7b7bee03 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Fri, 23 Sep 2011 20:25:32 -0400
Subject: [PATCH] media: handle dialog closing consistently

The 'Other media' dialog is just hidden when the close button is
clicked (so that it can be shown again), but when closing it with
the Escape key, it gets destroyed and an attempt to bring it up
again just shows a sad empty little square.

https://bugzilla.gnome.org/show_bug.cgi?id=659940

https://bugzilla.gnome.org/show_bug.cgi?id=659948
---
 panels/media/cc-media-panel.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/panels/media/cc-media-panel.c b/panels/media/cc-media-panel.c
index 96be39c..3f50068 100644
--- a/panels/media/cc-media-panel.c
+++ b/panels/media/cc-media-panel.c
@@ -409,6 +409,10 @@ on_extra_options_button_clicked (GtkWidget    *button,
                     "response",
                     G_CALLBACK (on_extra_options_dialog_response),
                     self);
+  g_signal_connect (dialog,
+                    "delete-event",
+                    G_CALLBACK (gtk_widget_hide_on_delete),
+                    NULL);
   /* update other_application_combo */
   other_type_combo_box_changed (GTK_COMBO_BOX (combo_box), self);
   gtk_window_present (GTK_WINDOW (dialog));
-- 
1.7.6.2